pkg/kubelet/prober/prober.go

```diff
@@ -231,11 +231,7 @@ func (p *prober) newExecInContainer(container api.Container, containerID kubecon
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
 		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false, nil)
-		if err != nil {
-			return nil, err
-		}
-
-		return buffer.Bytes(), nil
+		return buffer.Bytes(), err
```

Comment:
Actually, kube's `Cmd.CombinedOutput()` is 100% consistent with how go's `os/exec` package does it. You always get the stdout/stderr output in the `[]byte`, and err is either nil (no error) or set (something went wrong, including possibly a nonzero exit code).